src/libs/actions/Task.js

```diff
@@ -216,24 +219,29 @@ function completeTask(taskReportID, taskTitle, originalLastMessageText) {
     const failureData = [
         {
             onyxMethod: Onyx.METHOD.MERGE,
-            key: `${ONYXKEYS.COLLECTION.REPORT}${taskReportID}`,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`,
             value: {
                 stateNum: CONST.REPORT.STATE_NUM.OPEN,
                 statusNum: CONST.REPORT.STATUS.OPEN,
-                lastMessageText: originalLastMessageText,
+                lastMessageText: report.lastMessageText,
+                lastMessageHtml: report.lastMessageHtml,
+                lastVisibleActionCreated: report.lastVisibleActionCreated,
+                lastActorEmail: report.lastActorEmail,
+                lastActorAccountID: report.lastActorAccountID,
+                lastReadTime: report.lastReadTime,
             },
         },
         {
             onyxMethod: Onyx.METHOD.MERGE,
-            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${taskReportID}`,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${report.reportID}`,
             value: {[completedTaskReportAction.reportActionID]: {pendingAction: null}},
         },
     ];
 
     API.write(
         'CompleteTask',
         {
-            taskReportID,
+            taskReportID: report.reportID,
             completedTaskReportActionID: completedTaskReportAction.reportActionID,
         },
         {optimisticData, successData, failureData},
```

Comment:
NAB and not a change that was introduced by this PR but I think we can handle it while at it; successData is an empty array, let's remove it and not define it (do same for ReopenTask and EditTask) (you can refer to cancelTask where it's done right)

Reply:
Removed those empty `successData`